perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -793,68 +795,52 @@ def LoadWorkerOutput(output):
 
 
-def MultiStreamRWBenchmark(results, metadata, vms, command_builder,
-                           service, bucket_name, regional_bucket_name):
+def _RunMultiStreamProcesses(vms, command_builder, cmd_args,
+                             streams_per_vm, num_streams):
 
-  """A benchmark for multi-stream latency and throughput.
+  """Runs all of the multistream read or write processes and doesn't return
+     until they complete.
 
   Args:
-    results: the results array to append to.
-    metadata: a dictionary of metadata to add to samples.
-    vms: the VMs to run the benchmark on.
     command_builder: an APIScriptCommandBuilder.
-    service: an ObjectStorageService.
-    bucket_name: the primary bucket to benchmark.
-    regional_bucket_name: the secondary bucket to benchmark.
-
-  Raises:
-    ValueError if an unexpected test outcome is found from the API
-    test script.
+    cmd_args: arguments for the command_builder.
+    streams_per_vm: number of threads per vm.
+    num_streams: total number of threads to launch.
   """
 
-  logging.info('Starting multi-stream write test on %s VMs.', len(vms))
-
-  objects_written_file = posixpath.join(vm_util.VM_TMP_DIR,
-                                        OBJECTS_WRITTEN_FILE)
-
-  size_distribution = _DistributionToBackendFormat(
-      FLAGS.object_storage_object_sizes)
-  logging.info('Distribution %s, backend format %s.',
-               FLAGS.object_storage_object_sizes, size_distribution)
+  output = [None] * num_streams
 
-  streams_per_vm = FLAGS.object_storage_streams_per_vm
-  num_streams = streams_per_vm * len(vms)
-
-  def StartMultiStreamProcess(cmd_args, proc_idx, out_array):
+  def _StartMultiStreamProcess(proc_idx):
     vm_idx = proc_idx // streams_per_vm
     logging.info('Running on VM %s.', vm_idx)
     cmd = command_builder.BuildCommand(
         cmd_args + ['--stream_num_start=%s' % proc_idx])
     out, _ = vms[vm_idx].RobustRemoteCommand(cmd, should_log=True)
-    out_array[proc_idx] = out
-
-  def RunMultiStreamProcesses(command):
-    output = [None] * num_streams
-    # Each process has a thread managing it.
-    threads = [
-        threading.Thread(target=StartMultiStreamProcess,
-                         args=(command, i, output))
-        for i in xrange(num_streams)]
-    for thread in threads:
-      thread.start()
-    logging.info('Started %s processes.', num_streams)
-    for thread in threads:
-      thread.join()
-    logging.info('All processes complete.')
-    return output
+    output[proc_idx] = out
 
+  # Each process has a thread managing it.
+  threads = [
+      threading.Thread(target=_StartMultiStreamProcess, args=(i,))
+      for i in xrange(num_streams)]
+  for thread in threads:
+    thread.start()
+  logging.info('Started %s processes.', num_streams)
+  for thread in threads:
+    thread.join()
+  logging.info('All processes complete.')
+  return output
+
+
+def _MultiStreamWrites(results, metadata, vms, command_builder,
```

Comment:
Possible suggestion: merge the code in this function into `_RunMultiStreamProcesses`, adding a few more configuration parameters to that function (you may need to make an object holding the configuration just to make it all readable). That should simplify the read part of `MultiStreamRWBenchmark` as well as the write part. What do you think?